packages/webview_flutter/webview_flutter_platform_interface/lib/v4/src/webview_controller_delegate.dart

```diff
@@ -0,0 +1,286 @@
+  ///
+  /// Throws an ArgumentError if [key] is not part of the specified assets
+  /// in the pubspec.yaml file.
+  Future<void> loadFlutterAsset(
+    String key,
+  ) {
+    throw UnimplementedError(
+        'loadFlutterAsset is not implemented on the current platform');
+  }
+
+  /// Loads the supplied HTML string.
+  ///
+  /// The [baseUrl] parameter is used when resolving relative URLs within the
+  /// HTML string.
+  Future<void> loadHtmlString(
+    String html, {
+    String? baseUrl,
+  }) {
+    throw UnimplementedError(
+        'loadHtmlString is not implemented on the current platform');
+  }
+
+  /// Makes a specific HTTP request ands loads the response in the webview.
+  ///
+  /// [WebViewRequest.method] must be one of the supported HTTP methods
+  /// in [WebViewRequestMethod].
+  ///
+  /// If [WebViewRequest.headers] is not empty, its key-value pairs will be
+  /// added as the headers for the request.
+  ///
+  /// If [WebViewRequest.body] is not null, it will be added as the body
+  /// for the request.
+  ///
+  /// Throws an ArgumentError if [WebViewRequest.uri] has empty scheme.
+  Future<void> loadRequest(
+    LoadRequestParams params,
+  ) {
+    throw UnimplementedError(
+        'loadRequest is not implemented on the current platform');
+  }
+
+  /// Accessor to the current URL that the WebView is displaying.
+  ///
+  /// If no URL was ever loaded, returns `null`.
+  Future<String?> currentUrl() {
+    throw UnimplementedError(
+        'currentUrl is not implemented on the current platform');
+  }
+
+  /// Checks whether there's a back history item.
+  Future<bool> canGoBack() {
+    throw UnimplementedError(
+        'canGoBack is not implemented on the current platform');
+  }
+
+  /// Checks whether there's a forward history item.
+  Future<bool> canGoForward() {
+    throw UnimplementedError(
+        'canGoForward is not implemented on the current platform');
+  }
+
+  /// Goes back in the history of this WebView.
+  ///
+  /// If there is no back history item this is a no-op.
+  Future<void> goBack() {
+    throw UnimplementedError(
+        'goBack is not implemented on the current platform');
+  }
+
+  /// Goes forward in the history of this WebView.
+  ///
+  /// If there is no forward history item this is a no-op.
+  Future<void> goForward() {
+    throw UnimplementedError(
+        'goForward is not implemented on the current platform');
+  }
+
+  /// Reloads the current URL.
+  Future<void> reload() {
+    throw UnimplementedError(
+        'reload is not implemented on the current platform');
+  }
+
+  /// Clears all caches used by the [WebView].
+  ///
+  /// The following caches are cleared:
+  ///	1. Browser HTTP Cache.
+  ///	2. [Cache API](https://developers.google.com/web/fundamentals/instant-and-offline/web-storage/cache-api) caches.
+  ///    These are not yet supported in iOS WkWebView. Service workers tend to use this cache.
+  ///	3. Application cache.
+  ///	4. Local Storage.
```

Comment:
Let's take the opportunity of the breaking change to fix this oddity; lumping local storage in with caches is not great, since local storage can frequently be user data rather than cache. Let's make a new `clearLocalStorage`, and remove local storage from this list. Then later we'll have the flexibility to decide what public API we want to offer for clearing different kinds of things.

(/cc @bparrishMines)

Reply:
I am wondering if this is feasible on all platforms. For example on Android  we currently use the `clearCache` method from the [WebView SDK](https://developer.android.com/reference/android/webkit/WebView#clearCache(boolean)). This method accepts an additional parameter `includeDiskFiles` which clears the local storage. As far as I can see there is no separate `clearDiskFiles` method which allows for only clearing the local storage.

Apple provides the `WKWebsiteDataStore` class which we utilize to clear the cache, through the `[WKWebsiteDataStore removeDataOfTypes:modifiedSince:completionHandler:]` message (documentation can be found [here](https://developer.apple.com/documentation/webkit/wkwebsitedatastore/1532938-removedataoftypes?language=objc)). Here we can specify certain types of data to remove and Apple does allow us to distinguish between memory cache, local storage, cooklies, etc... So for Apple devices this should not be a problem.

Reply:
See https://github.com/flutter/plugins/pull/5086, which is why this was something I was thinking about in the review here. Local Storage and cache are totally different APIs on Android.

Reply:
I missed the separate API, thanks for informing me. I have added the additional method and matching test with commit  [789fd5a](https://github.com/flutter/plugins/pull/5109/commits/789fd5a26c7d75198b864dfef3412a6a6d659237).